internal/hcsoci/resources_wcow.go

```diff
@@ -147,6 +151,36 @@ func setupMounts(ctx context.Context, coi *createOptionsInternal, r *resources.R
 					return errors.Wrapf(err, "adding SCSI virtual disk mount %+v", mount)
 				}
 				r.Add(scsiMount)
+			} else if strings.HasPrefix(mount.Source, "sandbox://") {
+				// Mounts that map to a path in the UVM are specified with a 'sandbox://' prefix.
+				//
+				// Example: sandbox:///a/dirInUvm destination:C:\\dirInContainer.
+
+				// First convert to a path in the sandboxmounts path itself.
+				sandboxPath := convertToWCOWSandboxMountPath(mount.Source)
+
+				// Now we need to exec a process in the vm that will make these directories as theres
+				// no functionality in the Windows gcs to create an arbitrary directory.
+				//
+				// Create the directory, but also run dir afterwards regardless of if mkdir succeeded to handle the case where the directory already exists
+				// e.g. from a previous container specifying the same mount (and thus creating the same directory).
+				cmd := cmd.CommandContext(
+					ctx,
+					coi.HostingSystem,
+					"cmd",
+					"/c",
+					"mkdir",
+					sandboxPath,
+					"&",
+					"dir",
+					sandboxPath,
+				)
+				var b bytes.Buffer
+				cmd.Stderr = &b
+				cmd.Spec.User.Username = `NT AUTHORITY\SYSTEM`
+				if err := cmd.Run(); err != nil {
+					return errors.Wrapf(err, "failed to create sandbox mount directory in utility VM %q", b.String())
```

Comment:
Is there a reason we aren't using `ExecInHost` for this instead?

Reply:
`ExecInHost` takes a `*shimdiag.ExecProcessRequest` request which felt odd here.

Reply:
How so?

Reply:
I don't think we should have a mention of a shim in these packages/use anything from shimdiag, but we already have `ExecInHost` take this req in cmd so *shrugs*. The only thing `ExecInHost` buys us here really is setting the username so I don't feel a pressing need to swap but I can change if you feel strongly.

Reply:
hmm I see what you mean, but at the same time, it seems weird to me to repeat the logic here.

Reply:
Might be time for some refactoring of that code :)

Reply:
Agree

Reply:
Reopening this since I refactored that code

Reply:
Sure! I can change now